src/Command/ProbeDispatcherCommand.php

```diff
@@ -284,95 +280,66 @@ public function sendInstruction(array $instruction, $expectedRuntime = null): vo
      */
     private function handleResponse($channel, $data): void
     {
+        $startAt = microtime(true);
+        $bytes = strlen($data);
         $response = json_decode($data, true);
 
         if (!$response) {
-            $this->logger->warning('COMMUNICATION_FLOW: Response from worker could not be decoded to JSON.');
+            $this->logger->error('dispatcher: malformed json received from worker');
 
             return;
         }
 
-        if (!isset(
-            $response['type'],
-            $response['status'],
-            $response['body']['timestamp'],
-            $response['body']['contents'],
-            $response['debug']
-        )
-        ) {
-            $this->logger->error('COMMUNICATION_FLOW: Response ... was missing keys.');
+        if (!isset($response['pid'], $response['type'], $response['status'], $response['headers'], $response['contents'])) {
+            $this->logger->error('dispatcher: incomplete response received from worker');
+
+            return;
         }
 
+        $pid = $response['pid'];
         $type = $response['type'];
         $status = $response['status'];
-        //$timestamp = $response['body']['timestamp'];
-        /** @var array $contents */
-        $contents = $response['body']['contents'];
-        $debug = $response['debug'];
-        $pid = $debug['pid'];
-        $runtime = $debug['runtime'];
+        /** @var array|string $contents */
+        $contents = $response['contents'];
 
-        $this->logger->info("COMMUNICATION_FLOW: Master received $type response from worker $pid with a runtime of $runtime.");
+        $this->logger->info("dispatcher: $type response ($bytes bytes) received from worker $pid");
 
         switch ($type) {
-            case 'exception':
```

Comment:
We should still do this ?

Reply:
Workers already log these themselves. It just leads to logging the same thing twice.

Reply:
Ok if an empty response is sent